huaweicloud/services/dcs/resource_huaweicloud_dcs_instance.go

```diff
@@ -1160,12 +1179,38 @@ func resourceDcsInstancesDelete(ctx context.Context, d *schema.ResourceData, met
 
 	// for prePaid mode, we should unsubscribe the resource
 	if d.Get("charging_mode").(string) == chargeModePrePaid {
-		err = common.UnsubscribePrePaidResource(d, cfg, []string{d.Id()})
+		retryFunc := func() (interface{}, bool, error) {
+			err = common.UnsubscribePrePaidResource(d, cfg, []string{d.Id()})
+			retry, err := handleOperationError(err)
+			return nil, retry, err
+		}
+		_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
```

Comment:
build retryfunc in the if..else, and move the common.Retry out

Reply:
done